### ISSUE TITLE
getOM.sh: java 8 update site not selected by default

### ISSUE THREAD

**Issue by cpoczatek** [BWHCNI/OpenMIMS#44]
On a fresh install the java 8 update site not checked by default. I think this is required of OM now. getOM.sh should add this update site.

**Comment by cpoczatek** [comment]
I got this kinda backwards. A fresh install with getOM.sh errors our with:

```
(Fiji Is Just) ImageJ 2.0.0-rc-43/1.51g; Java 1.6.0_24 [64-bit]; Linux 4.4.0-38-generic; 39MB of 2805MB (1%)
java.lang.UnsupportedClassVersionError: com/nrims/NRIMS_Plugin : Unsupported major.minor version 52.0
This plugin requires Java 1.8 or later.
```

However adding the java8 update site errors out with:

```
java.lang.UnsupportedClassVersionError: net/imagej/Main : Unsupported major.minor version 52.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClassCond(ClassLoader.java:632)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:616)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
    at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:307)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:248)
    at net.imagej.launcher.ClassLauncher.launch(ClassLauncher.java:252)
    at net.imagej.launcher.ClassLauncher.run(ClassLauncher.java:186)
    at net.imagej.launcher.ClassLauncher.main(ClassLauncher.java:77)
Falling back to remote updater at http://update.imagej.net/bootstrap.js
java.lang.UnsupportedClassVersionError: org/scijava/plugins/scripting/java/JavaScriptLanguage : Unsupported major.minor version 52.0
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClassCond(ClassLoader.java:632)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:616)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
    at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
    at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:307)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:248)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:247)
    at sun.misc.Service$LazyIterator.next(Service.java:271)
    at javax.script.ScriptEngineManager.initEngines(ScriptEngineManager.java:109)
    at javax.script.ScriptEngineManager.access$000(ScriptEngineManager.java:37)
    at javax.script.ScriptEngineManager$1.run(ScriptEngineManager.java:80)
    at java.security.AccessController.doPrivileged(Native Method)
    at javax.script.ScriptEngineManager.init(ScriptEngineManager.java:78)
    at javax.script.ScriptEngineManager.<init>(ScriptEngineManager.java:51)
    at net.imagej.launcher.RemoteUpdater.runAsJavascript(RemoteUpdater.java:86)
    at net.imagej.launcher.RemoteUpdater.runRemote(RemoteUpdater.java:71)
    at net.imagej.launcher.ClassLauncher.run(ClassLauncher.java:189)
    at net.imagej.launcher.ClassLauncher.main(ClassLauncher.java:77)
```

**Comment by cpoczatek** [comment]
Adding a line to runOpenMIMS.sh post-install to use the system jre fixes this:

```
JHOME=$(which java) #new line

./ImageJ-linux64 \
--java-home $JHOME #new arg

```

Why isn't Fiji installing a local java 8 jre?

**Comment by taylorwrt** [comment]
Collin,

Thanks for your help.  I just saw your emails tonight at 11:30, so I will look into them in detail tomorrow.

Walter

**Comment by taylorwrt** [comment]
Collin,

Under what circumstances is this shell script used?   Maybe a dumb question, but ….

On a Mac, double clicking on the Fiji application icon launches Fiji and OpenMIMS (via the AutoRun_Scripts.ijm file, I assume).
Just for jollies, I did a cd to the lib folder inside Fiji and ran the script, but that fails (also fails on Linux).

Also, removing everything from the lib folder does not seem to prevent Fiji from auto-running OpenMIMS, so I do not see where this script comes into play.

Walter

**Comment by cpoczatek** [comment]
getOM.sh is an [installer script for linux](https://github.com/BWHCNI/OpenMIMS/wiki/Installation). I was checking things with segmentation and wanted a fresh install. This used to work... :(

**Comment by taylorwrt** [comment]
Collin,

When would getOM.sh get invoked?   When I installed Fiji and configured it on Linux, once on my laptop and once on my home machine, I never invoked this script, at least not knowingly.   

Looking at it, it seems that it combines all the steps in getting and configuring Fiji to run OpenMIMS, perhaps simplifying things, but not actually being required.  Is that correct?

Is this script intended for the average user of OpenMIMS, or only advanced users?

In any case, I am going through it to make it platform independent, and fix the problems you found.   Also to get better acquainted with shell scripting, which I never learned to any great level.   It seems useful.  Kinda like old-fashioned basic with powerful Unix commands easily accessible.

Walter

**Comment by taylorwrt** [comment]
Collin,

I have been fighting with this script for days now  (on Linux).    A summary:

I was wondering why your curl got Fiji from the Jenkins site.  I changed it to this, since it seemed like the “official” place to get Fiji.

curl http://downloads.imagej.net/fiji/latest/fiji-linux64.zip > fiji-linux64.zip

This version seems to have Java 1.8 embedded in it.

This did not fix  the error about the missing libjvm.so, though.

If I include the —java-home $JHOME thing, it always looks for libjvm.so   in a place where it does not exist (/usr/bin/java/lib/amd64/server/)

If I leave that parameter off, it launches Fiji, but does not automatically launch OpenMIMS, but I also do not get any errors.  Maybe that is OK?
I get this:

 ./Fiji.app/ImageJ-linux64 
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
Java HotSpot(TM) 64-Bit Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release

In Fiji, the OpenMIMS plugin is there, and the proper version.  So is this the desired outcome?

Walter

**Comment by taylorwrt** [comment]
I am closing this for the same reason I closed bug #43:  We no longer recommend the use of curl to download either Fiji or OpenMIMS. Since the update site for OpenMIMS is now hosted in imageJ, like most of the plugins, rather than on a Partners virtual machine, the plugin for OpenMIMS is always available in freshly-downloaded copies of Fiji. Now to get OpenMIMS, one only needs to go to the Help menu of Fiji and select "Update..." then click on "Manage update sites" button and check the entry for OpenMIMS. After that, the "Open MIMS Image" item will be available on the Fiji Plugins menu, and will updated any time the user goes to the Update menu and clicks on "Apply changes". This seems to me simpler than the use of curl on the command line. I have updated the manual to recommend this technique rather than curl. No command line stuff, no custom shell scripts that always seem to break, no confusing symlinks, and no worrying that Partners will turn off the virtual machine that was hosting update site.
